test/test_dynamic_shapes.py

```diff
@@ -569,18 +545,30 @@ def test_method(self, fn, first_type, second_type):
         if is_unary_fn and second_type == "float":
             self.skipTest(f"{fn} is unary and already tested")
 
-        # We could pass int/float directly for types but then the
-        # mangled test name is bad
-        inp1 = random.random() * 2.5
-        if first_type == "int":
-            inp1 = int(inp1)
-        inp2 = random.random() * 2.5
-        if second_type == "int":
-            inp2 = int(inp2)
+        # Only floats here since these will be converted to int if necessary.
+        # We also ignore complex and bool.
+        values = (
+            0.0,
+            1.0,
+            2.5,
+        )
 
-        shape_env = ShapeEnv()
+        neg_values = tuple(-x for x in values)
```

Comment:
Added more inputs since these create interesting corner cases: short-circuit behavior in some ops, complex output, etc.